src/main/java/test/matrix/MatrixAndWordImpl.java

```diff
@@ -0,0 +1,135 @@
+package test.matrix;
+
+import java.util.NoSuchElementException;
+import test.matrix.exceptions.EmptyInputException;
+import test.matrix.exceptions.IncorrectLengthException;
+
+public class MatrixAndWordImpl implements MatrixAndWord {
+    private int firstPosition;
+    private int secondPosition;
+    private int size;
+    private int lengthOfMatrix;
+    private String word;
+    private String letters;
+    private String firstLetterPosition;
+    private Character[][] characters;
+
+    @Override
+    public String getSequence_Of_Cells(String letters, String word) {
+        this.letters = letters.toUpperCase();
+        this.word = word.toUpperCase();
+        size = 0;
+        checkInputData();
+        lengthOfMatrix = (int) Math.sqrt(this.letters.length());
+        characters = new Character[lengthOfMatrix][lengthOfMatrix];
+        getCharacterArray();
+        StringBuilder result = new StringBuilder();
+        result.append(getFirstLetter());
+        return finder(result);
+    }
+
+    private String finder(StringBuilder result) {
+        int counter = 0;
+        boolean isFoundLetter = false;
+        while (counter < this.word.length()) {
+            for (int i = firstPosition != 0 ? firstPosition - 1 : 0; i < lengthOfMatrix; i++) {
+                for (int j = secondPosition != 0 ? secondPosition - 1 : 0;
+                        j < lengthOfMatrix; j++) {
+                    if (size < this.word.length()
+                            && characters[i][j] == this.word.charAt(size)
+                            && (((i == firstPosition + 1 || i == firstPosition - 1)
+                            && j == secondPosition)
+                            || ((j == secondPosition + 1 || j == secondPosition - 1)
+                            && i == firstPosition))
+                            && checkNextLetters(i, j, size)) {
```

Comment:
this check seems to be complicated, let's create method for it, it will improve readability